bazel/external/http-parser.BUILD

```diff
@@ -7,6 +7,7 @@ cc_library(
         "http_parser.h",
     ],
     hdrs = ["http_parser.h"],
+    copts = ["-DHTTP_MAX_HEADER_SIZE=0x7fffffff"],
```

Comment:
Can you add a comment on why we set this and why it's safe (we have checks elsewhere, etc.)

Reply:
Also, don't you need HTTP/1 codec changes now to check the current header map size on each new header as we do with HTTP/2? I'm not sure how this is safe otherwise?

Reply:
done on the comment, working on the http1 codec check

Reply:
Also as an aside, when this lands please ping the googlers on current and next import duty as this will almost certainly need custom tweaks on import.

Reply:
Actually sorry, on second thought I'm having a moment of paranoia.  We don't have anywhere else in the Envoy code base where we use http_parser so this is 100% OK today but what do we think of adding a limit just in case?  If we added a 200KB limit or some such it'd be very safely above our window such that we should never hit the http_parser limits in the codec, without allowing total infinite memory if someone else uses http parser.
Thoughts?